include/tapkee/routines/landmarks.hpp

```diff
@@ -0,0 +1,81 @@
+/* This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Copyright (c) 2012-2024 Sergey Lisitsyn
+ */
+#pragma once
+
+/* Tapkee includes */
+#include <tapkee/defines.hpp>
+#include <tapkee/utils/time.hpp>
+/* End of Tapkee includes */
+
+namespace tapkee
+{
+namespace tapkee_internal
+{
+
+template <class RandomAccessIterator>
+Landmarks select_landmarks_random(RandomAccessIterator begin, RandomAccessIterator end, ScalarType ratio)
+{
+    Landmarks landmarks;
+    landmarks.reserve(end - begin);
+    for (RandomAccessIterator iter = begin; iter != end; ++iter)
+        landmarks.push_back(iter - begin);
+    tapkee::random_shuffle(landmarks.begin(), landmarks.end());
+    landmarks.erase(landmarks.begin() + static_cast<IndexType>(landmarks.size() * ratio), landmarks.end());
+    return landmarks;
+}
+
+template <class RandomAccessIterator, class PairwiseCallback>
+DenseMatrix triangulate(RandomAccessIterator begin, RandomAccessIterator end, PairwiseCallback distance_callback,
+                        Landmarks& landmarks, DenseVector& landmark_distances_squared,
+                        EigendecompositionResult& landmarks_embedding, IndexType target_dimension)
+{
+    timed_context context("Landmark triangulation");
+
+    const IndexType n_vectors = end - begin;
+    const IndexType n_landmarks = landmarks.size();
+
+    bool* to_process = new bool[n_vectors];
+    std::fill(to_process, to_process + n_vectors, true);
```

Comment:
I think these two are also possible with `std::vector to_process(n_vectors, true)`.
Then, it would be a `std::vector<bool>`, so, still living dangerously 🤪

Reply:
Absolutely

Reply:
@iglesias fixed that

Reply:
Thank you.

I think thanks to CTAD it should be enough just with `std::vector`, deducing `bool` from the second argument to the constructor. Mentioning it just for the discussion, not implying it needs to be changed :) here it is a syntactic detail, but can be nice to know.